lib/versioning/VersioningRequestProcessor.js

```diff
@@ -50,8 +50,8 @@ class VersioningRequestProcessor {
         this.repairing = {};
     }
 
-    generateVersionId() {
-        return genVID(this.uidCounter++);
+    generateVersionId(replicationGroupToken) {
+        return genVID(this.uidCounter++, replicationGroupToken);
```

Comment:
You can get the thing you want from config and use it here without the need to get it from input. Then you can keep the rest unchanged. For PHD versions, you can generate the versionId here then pass it to the function `generatePHDVersion()`, making Version.js independent from VersionID.js.